### PR TITLE
Issue #221:  Also check for errno 35

### PR DESCRIPTION
**Pull request by palladia**

### CODE REVIEW

**Comment by palladia** [comment]
@Microsoft/ostc-devs

File: Unix/tests/http/test_http.cpp [hunk 1]
```diff
@@ -730,12 +732,14 @@ static void _ConnectToServerExpectConnectionDrop(const string& data, MI_Uint32 s
...
         size_t read = 0;
+        int err;
 
         do
         {
             r = Sock_Read(sock, r_buf, sizeof(r_buf), &read);
+            err = Sock_GetLastError();
         }
-        while (r != MI_RESULT_OK && Sock_GetLastError() == 11 /*EAGAIN*/);
+        while (r != MI_RESULT_OK && (err == 11 || err == 35) /*EAGAIN*/);
```
Comment (jeffaco) [inline Unix/tests/http/test_http.cpp]: Err 11 is EGAGIN, not 35. Can you clarify what 35 is, here and below?

File: Unix/tests/http/test_httpclient.cpp [hunk 2]
```diff
@@ -395,12 +395,14 @@ static void* MI_CALL _http_server_proc(void* param)
...
     size_t read = 0;
+    int err;
 
     do
     {
         r = Sock_Read(sock, r_buf, sizeof(r_buf), &read);
+        err = Sock_GetLastError();
     }
-    while (r != MI_RESULT_OK && Sock_GetLastError() == 11 /*EAGAIN*/);
+    while (r != MI_RESULT_OK && (err == 11 || err == 35) /*EAGAIN*/);
```
Comment (jeffaco) [inline Unix/tests/http/test_httpclient.cpp]: Ditto to above, this is wrong from a comment perspective.

File: Unix/tests/wsman/test_wsmanredirect.cpp [hunk 3]
```diff
@@ -107,12 +107,15 @@ static void* MI_CALL _http_server_proc(void* param)
...
-    
+    int err;
+
     do
     {
         r = Sock_Read(sock, r_buf, sizeof(r_buf), &read);
+        err = Sock_GetLastError();
     }
-    while (r != MI_RESULT_OK && Sock_GetLastError() == 11 /*EAGAIN*/);
+    while (r != MI_RESULT_OK && (err == 11 || err == 35) /*EAGAIN*/);
```
Comment (jeffaco) [inline Unix/tests/wsman/test_wsmanredirect.cpp]: and ditto ...

File: Unix/tests/http/test_http.cpp (hunk 1 above)
Comment (paulcallen) [inline Unix/tests/http/test_http.cpp]: I believe if you include errno.h you can use the actual defines for the different errors which is more explanitory

Comment (jeffaco) [inline Unix/tests/http/test_http.cpp]: That's actually much better (use the #define value). Allows for self-documenting code ...

File: Unix/tests/http/test_http.cpp [hunk 4]
```diff
@@ -185,14 +185,16 @@ static void* MI_CALL _http_client_proc(void* param)
...
     size_t read = 0;
+    int err;
 
     do
     {
         r = Sock_Read(sock, r_buf, sizeof(r_buf), &read);
+        err = Sock_GetLastError();
     }
-    while (r != MI_RESULT_OK && Sock_GetLastError() == 11 /*EAGAIN*/);
+    while (r != MI_RESULT_OK && (err == 11 || err == 35) /*EAGAIN and EDEADLK*/);
```
Comment (jeffaco) [inline Unix/tests/http/test_http.cpp]: This isn't quite what I had in mind.

Can't you include errlog.h and then do something like:

`while (r != MI_RESULT_OK && (err == EAGAIN || err == EDEADLK));`

That's more self documenting ...

Comment (palladia) [inline Unix/tests/http/test_http.cpp]: I don't think error codes (from errno.h) is standardized for all platforms.  What's confusing is that errno of 35 could be EAGAIN, ENOMSG, EDEADLK, or undefined, just on a few systems that I checked.

Comment (jeffaco) [inline Unix/tests/http/test_http.cpp]: That doesn't make sense, especially for the lower numbered (standard) error codes.

Can you please point out two systems that conflict with one another so I can take a look, thanks.

**Comment by palladia** [comment]
@jeffaco Please re-review.

**Pull request merged**